app/mailers/application_mailer.rb

```diff
@@ -1,11 +1,29 @@
 require 'uri'
+require 'shellwords'
 
 class ApplicationMailer < ActionMailer::Base
   include Roadie::Rails::Automatic
   default :delivery_method => proc { Setting[:delivery_method] },
           :from => proc { Setting[:email_reply_address] || "noreply@foreman.example.org" }
   after_action :set_delivery_options
 
+  def self.delivery_settings
+    options = {}
+    Foreman.settings.categories['email'].each do |setting|
+      ["smtp", "sendmail"].each do |method|
+        if Setting[:delivery_method].to_s == method && setting.name.start_with?(method) && setting.value.to_s.present?
+          opt_name = name.to_s.gsub(/^#{method}_/, '')
+          if setting.name == "sendmail_arguments"
+            options[opt_name] = Shellwords.shellescape(setting.value)
+          else
+            options[opt_name] = setting.value
+          end
+        end
+      end
+    end
```

Comment:
I had some real difficulty reading this due to the additional loop. How about this:
```suggestion
    prefix = "#{Setting[:delivery_method]}_"

    Foreman.settings.categories['email'].each do |setting|
      if setting.name.start_with?(prefix) && setting.value.to_s.present?
        opt_name = name.delete_prefix(prefix)
        options[opt_name] = if setting.name == "sendmail_arguments"
                              Shellwords.shellescape(setting.value)
                            else
                              setting.value
                            end
      end
    end

```

Perhaps this needs an early `return {} unless ['smtp', 'sendmail'].include?(Setting[:delivery_method])` but I didn't check those are the only two options and it wouldn't fail otherwise.

Reply:
it should not be anything else, but to be on the safe side let's have that check :)